### ISSUE TITLE
mentions list is NOT same as in twister-html

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [swizzler/swizzler#7]
1. see pic: in twister-html mentions are shown
2. OK is that: 
   a. direct messages and 
   b. home timeline are same as in twister-html
3. observation: restarted browser (but let swizzler still active) and one mention is shown now
   though it's an old one (from 2014-05-12 09:09:58)

question: is this function working, but "only" slow?

![mention3](https://cloud.githubusercontent.com/assets/642286/2996345/c4085188-dcec-11e3-9599-cf34d6c29fc0.png)

**Comment by thedod** [comment]
TL;DR - This is a known bug I hope to solve once we move from `twister.py` to `pytwister` (a separate project coming soon. Traveling at the moment).

[@miguelfreitas, do I understand this correctly?]
The problem is that due to the nature of udp, you can sometimes receive no mentions, several sporadic mentions from months ago, or an up to date mention stream with only one or two twists missing.
I think that the way the html client handles this problem is by saving whatever twists it gets into local storage. This means that if you had access to some mentions yesterday, today they can be retrieved from cache even if queries to twisterd return empty results.

At the moment, twister.py is mostly a cached interface to twisterd, but the cache expires pretty fast (because it's hard to make the cache ignore some results it should), and the whole things needs to be rewritten with a proper local storage.
Pytwister won't be a thin cached wrapper around twisterd, but a type-aware [encrypted database](http://peewee.readthedocs.org/en/latest/peewee/playhouse.html#sqlcipher-ext). It will not only be used by swizzler, but can be used as a high level API by other client (I'm toying with the idea of writing an [ncurses](http://npyscreen.readthedocs.org/) one.

I leave this issue open, of course, but it would probably take a while until I can fix this.

**Comment by Erkan-Yilmaz** [comment]
@thedod thx for the detailed explanation
perhaps the page could have an additional explanation to help the user understand better?
e.g.
Nothing... (the twister cache may have expired too fast)

**Comment by thedod** [comment]
You're right.
As a temporary measure, I'll rephrase for "found nothing" messages to something that explains the problem (maybe even a link to a page in the project's wiki and then there's no limit on text size)

**Comment by miguelfreitas** [comment]
@thedod i didn't checked how you do in swizzler, but you should also try playing with additional dhtget parameters. besides the cache you have already mentioned, twister-html also waits for more peers to answer the request:

[10000,2000,3]); // use extended timeout parameters (requires twister_core >= 0.9.14)

that is: wait up to 10 seconds total, with 2 seconds timeout between replies, wait for at least 3 replies (from different peers). this way you should collect more answers.

not all peers may have the entire set of mentions and the fastest one isn't necessarily the most complete.

**Comment by thedod** [comment]
Thanks.
I'll try to give it a go.
